compute/instances/custom-machine-type/helper_class.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"fmt"
+	"strings"
+)
+
+// [START compute_custom_machine_type_helper_class]
+
+const (
+	N1       = "custom"
+	N2       = "n2-custom"
+	N2D      = "n2d-custom"
+	E2       = "e2-custom"
+	E2Micro  = "e2-custom-micro"
+	E2Small  = "e2-custom-small"
+	E2Medium = "e2-custom-medium"
+)
+
+type TypeLimit struct {
+	allowedCores     []int
+	minMemPerCore    int
+	maxMemPerCore    int
+	allowExtraMemory bool
+	extraMemoryLimit int
+}
+
+var (
+	CPUSeriesE2Limit       = TypeLimit{MakeRange(2, 33, 2), 512, 8192, false, 0}
+	CPUSeriesE2MicroLimit  = TypeLimit{[]int{}, 1024, 2048, false, 0}
+	CPUSeriesE2SmallLimit  = TypeLimit{[]int{}, 2048, 4096, false, 0}
+	CPUSeriesE2MeidumLimit = TypeLimit{[]int{}, 4096, 8192, false, 0}
+	CPUSeriesN2Limit       = TypeLimit{append(MakeRange(2, 33, 2), MakeRange(36, 129, 4)...), 512, 8192, true, 624 << 10}
+	CPUSeriesN2DLimit      = TypeLimit{[]int{2, 4, 8, 16, 32, 48, 64, 80, 96}, 512, 8192, true, 768 << 10}
+	CPUSeriesN1Limit       = TypeLimit{append([]int{1}, MakeRange(2, 97, 2)...), 922, 6656, true, 624 << 10}
+)
+
+type CustomMachineType struct {
+	zone, cpuSeries     string
+	memoryMb, coreCount int
+	typeLimit           TypeLimit
+}
+
+func (t CustomMachineType) check() error {
```

Comment:
I wonder if there's a better name than check. Maybe validate? Also, would this be better as a function rather than a method?